test/unit/infrastructure/adapters/GitCasWarpStateCacheAdapter.retention.test.ts

```
@@ -0,0 +1,311 @@
+  ref: 'refs/cas/rootsets/git-warp/demo/state-cache',
+  headOid: rootEntries.length === 0 ? null : ROOT_HEAD,
+  treeOid: rootEntries.length === 0 ? null : ROOT_HEAD,
+  entries: [...rootEntries],
+}));
+const mockRootMutate = vi.fn(async (
+  mutator: (
+    entries: ReadonlyArray<Readonly<RootSetEntry>>,
+  ) => Iterable<RootSetEntry> | Promise<Iterable<RootSetEntry>>,
+): Promise<RootSetMutationResult> => {
+  events.push('root:prepare');
+  rootEntries = Array.from(await mutator(rootEntries));
```

ESLint

[error] Use spread syntax [...iterable] instead of Array.from(iterable) when no mapper function is provided

(e18e/prefer-spread-syntax)